libs/SmartSync/SmartSync/Classes/Util/SFMruSyncDownTarget.m

```diff
@@ -87,14 +87,15 @@ - (void) startFetch:(SFSmartSyncSyncManager*)syncManager
     __weak SFMruSyncDownTarget *weakSelf = self;
     SFRestRequest *request = [[SFRestAPI sharedInstance] requestForMetadataWithObjectType:self.objectType];
     [SFSmartSyncNetworkUtils sendRequestWithSmartSyncUserAgent:request failBlock:errorBlock completeBlock:^(NSDictionary* d) {
-        NSArray* recentItems = [weakSelf pluck:d[kRecentItems] key:self.idFieldName];
+        __strong typeof(self) strongSelf = weakSelf;
+        NSArray* recentItems = [strongSelf pluck:d[kRecentItems] key:self.idFieldName];
         NSString* inPredicate = [@[ self.idFieldName, @" IN ('", [recentItems componentsJoinedByString:@"', '"], @"')"]
                                  componentsJoinedByString:@""];
         NSString* soql = [[[[SFSmartSyncSoqlBuilder withFieldsArray:self.fieldlist]
```

Comment:
Good catch - I did weakSelf->strongSelf string replacements but didn't check for self.